tlsutil/config.go

```diff
@@ -455,12 +456,23 @@ func newX509CertPool(groups ...[]string) (*x509.CertPool, error) {
 			}
 		}
 	}
-	if len(pool.Subjects()) == 0 {
+	if !haveSubjects(pool) {
```

Comment:
I think I would actually prefer if we drop this conditional entirely so we don't need this reflect trickery in non-test code.

Reply:
cc: @i0rek  who might have more context on what this was for originally in #5597

Reply:
If we're only worried that a cert was added and not that it had a valid name, this might work:
```golang
func newX509CertPool(groups ...[]string) (*x509.CertPool, error) {
	var haveCerts bool
	pool := x509.NewCertPool()
	for _, group := range groups {
		for _, pem := range group {
			if !pool.AppendCertsFromPEM([]byte(pem)) {
				return nil, fmt.Errorf("failed to parse PEM %s", pem)
			}
			if len(pem) > 0 {
				haveCerts = true
			}
		}
	}
	if !haveCerts {
		return nil, nil
	}
	return pool, nil
}
```